cinema/serializers.py

```diff
@@ -59,6 +68,11 @@ class MovieSessionListSerializer(MovieSessionSerializer):
     cinema_hall_capacity = serializers.IntegerField(
         source="cinema_hall.capacity", read_only=True
     )
+    # tickets_available = serializers.IntegerField(read_only=True)
+    tickets_available = serializers.SerializerMethodField()
+
+    def get_tickets_available(self, obj):
+        return obj.cinema_hall.capacity - obj.tickets.count()
```

Comment:
This is quiet inefficient because we will have to query database for tickets.count() on every entry in a list. Better to override queryset and use annotate()